File: implementation/scipamato/common/wicket/src/test/java/ch/difty/scipamato/common/web/pages/login/AbstractLogoutPageTest.java
package ch.difty.scipamato.common.web.pages.login;

import de.agilecoders.wicket.core.markup.html.bootstrap.common.NotificationPanel;
import org.apache.wicket.markup.html.form.StatelessForm;
import org.apache.wicket.request.mapper.parameter.PageParameters;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.mock.mockito.MockBean;

import ch.difty.scipamato.common.DateTimeService;
import ch.difty.scipamato.common.config.ApplicationProperties;
import ch.difty.scipamato.common.web.AbstractPage;
import ch.difty.scipamato.common.web.WicketBaseTest;

class AbstractLogoutPageTest extends WicketBaseTest {

    private AbstractLogoutPage page;

    @MockBean
    private ApplicationProperties applicationProperties;
    @MockBean
    private DateTimeService       dateTimeService;

    @Override
    protected void setUpHook() {
        page = new TestLogoutPage(new PageParameters());
    }

    @Test
    void assertPage() {
        getTester().startPage(page);
        getTester().assertRenderedPage(AbstractLogoutPage.class);

        getTester().assertInvisible("navbar");
        getTester().assertComponent("feedback", NotificationPanel.class);
        getTester().assertComponent("form", StatelessForm.class);
    }

    @Test
    void submit_withResponsePage() {
        AbstractLogoutPage page2 = new TestLogoutPage(new PageParameters()) {
            @Override
            protected AbstractPage<?> getResponsePage() {
                return new TestLoginPage(new PageParameters());
            }
        };
        getTester().startPage(page2);
        getTester().assertRenderedPage(AbstractLogoutPage.class);
        getTester().submitForm("form");

        getTester().assertRenderedPage(TestLoginPage.class);
        getTester().assertNoErrorMessage();
    }

    @Test
    void submitting_withoutLogoutData_rendersErrorMessages() {
        getTester().startPage(page);
        getTester().assertRenderedPage(AbstractLogoutPage.class);
        getTester().submitForm("form");

        getTester().assertRenderedPage(AbstractLogoutPage.class);

        getTester().assertNoErrorMessage();
    }

}

File: implementation/scipamato/common/wicket/src/test/kotlin/ch/difty/scipamato/common/web/pages/login/AbstractLogoutPageTest.kt
package ch.difty.scipamato.common.web.pages.login

import ch.difty.scipamato.common.DateTimeService
import ch.difty.scipamato.common.config.ApplicationProperties
import ch.difty.scipamato.common.web.AbstractPage
import ch.difty.scipamato.common.web.WicketBaseTest
import de.agilecoders.wicket.core.markup.html.bootstrap.common.NotificationPanel
import org.apache.wicket.markup.html.form.StatelessForm
import org.apache.wicket.request.mapper.parameter.PageParameters
import org.junit.jupiter.api.Test
import org.springframework.boot.test.mock.mockito.MockBean

internal class AbstractLogoutPageTest : WicketBaseTest() {

    private lateinit var page: AbstractLogoutPage<*>

    @MockBean
    private lateinit var applicationProperties: ApplicationProperties

    @MockBean
    private lateinit var dateTimeService: DateTimeService

    override fun setUpHook() {
        page = TestLogoutPage(PageParameters())
    }

    @Test
    fun assertPage() {
        tester.startPage(page)
        tester.assertRenderedPage(AbstractLogoutPage::class.java)

        tester.assertInvisible("navbar")
        tester.assertComponent("feedback", NotificationPanel::class.java)
        tester.assertComponent("form", StatelessForm::class.java)
    }

    @Test
    fun submit_withResponsePage() {
        val parameters = PageParameters()
        val page2 = object : TestLogoutPage(parameters) {
            override fun getResponsePage(): AbstractPage<*> = TestLoginPage(parameters)
        }
        tester.startPage(page2)
        tester.assertRenderedPage(AbstractLogoutPage::class.java)
        tester.submitForm("form")

        tester.assertRenderedPage(TestLoginPage::class.java)
        tester.assertNoErrorMessage()
    }

    @Test
    fun submitting_withoutLogoutData_rendersErrorMessages() {
        tester.startPage(page)
        tester.assertRenderedPage(AbstractLogoutPage::class.java)
        tester.submitForm("form")

        tester.assertRenderedPage(AbstractLogoutPage::class.java)

        tester.assertNoErrorMessage()
    }
}
